tests/trainer/test_trainer.py

```diff
@@ -2161,40 +2161,40 @@ def training_step(self, batch, batch_idx):
         (
             dict(strategy="ddp_fully_sharded", gpus=1),
             dict(
-                _distrib_type=DistributedType.DDP_FULLY_SHARDED,
+                _distrib_type=_StrategyType.DDP_FULLY_SHARDED,
                 _device_type=DeviceType.GPU,
                 num_gpus=1,
                 num_processes=1,
             ),
         ),
         (
             dict(strategy=DDPSpawnPlugin(), num_processes=2, gpus=None),
-            dict(_distrib_type=DistributedType.DDP_SPAWN, _device_type=DeviceType.CPU, num_gpus=0, num_processes=2),
+            dict(_distrib_type=_StrategyType.DDP_SPAWN, _device_type=DeviceType.CPU, num_gpus=0, num_processes=2),
         ),
         (
             dict(strategy=DDPSpawnPlugin(), gpus=2),
-            dict(_distrib_type=DistributedType.DDP_SPAWN, _device_type=DeviceType.GPU, num_gpus=2, num_processes=1),
+            dict(_distrib_type=_StrategyType.DDP_SPAWN, _device_type=DeviceType.GPU, num_gpus=2, num_processes=1),
         ),
         (
             dict(strategy=DDPPlugin(), num_processes=2, gpus=None),
-            dict(_distrib_type=DistributedType.DDP, _device_type=DeviceType.CPU, num_gpus=0, num_processes=2),
+            dict(_distrib_type=_StrategyType.DDP, _device_type=DeviceType.CPU, num_gpus=0, num_processes=2),
```

Comment:
All these accesses in our tests will trigger deprecation warnings.

This goes against #9939 

Can you update all test usages to use the non-deprecated alternative (str) or catch the warnings with `pytest.deprecated_call(match=...)`?